lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Hunspell.java

```diff
@@ -540,7 +566,36 @@ boolean acceptsStem(int formID) {
     if (word.contains("-") && suggestions.stream().noneMatch(s -> s.contains("-"))) {
       suggestions.addAll(modifyChunksBetweenDashes(word));
     }
+  }
+
+  private Runnable checkTimeLimit(String word, WordCase wordCase, Set<String> suggestions) {
+    return new Runnable() {
+      final long limit = System.nanoTime() + TimeUnit.MILLISECONDS.toNanos(SUGGEST_TIME_LIMIT);
```

Comment:
What happens in the case of overflow here, the javadocs for `nanotime()` state:
> one should use t1 - t0 < 0, not t1 < t0, because of the possibility of numerical overflow.

Reply:
TIL, thanks!